src/server/editorServices.ts

```diff
@@ -1664,6 +1664,13 @@ namespace ts.server {
                     const jsconfigFileName = asNormalizedPath(combinePaths(searchPath, "jsconfig.json"));
                     result = action(jsconfigFileName, combinePaths(canonicalSearchPath, "jsconfig.json"));
                     if (result) return jsconfigFileName;
+
+                    // If we started within node_modules, don't look outside node_modules.

```

Comment:
Can you please add tests so we have coverage for this.

Reply:
Sorry, this was originally a draft PR to get feedback and I forgot to add them when I promoted it.  Will do.